Ec dh general definitions doc (#3158)
mendix/docs: content/data-hub/general/data-hub-concepts.md

---
title: "Concepts behind Mendix Data Hub and Data Sharing (Working title)"
category: "General Info"
menu_order: 12
description: "Describes terminology and concepts"
tags: ["data hub catalog", "data hub", "virtual entities", "landscape", "published odata service"]

---

- [ ]  Add discoverability and validity

# 1 Introduction

This document describes the concepts behind using the Mendix Data Hub and sharing datasets in your app development.

Catalog allows governance of which data is used by whom, and life cycle management informing people of changes, and with good versioning of OData contracts.

##  1.1 The Data Hub Components

> Data Hub is made up of the following components:

Reviewer: These are parts of the Catalog UI. The Data Hub includes the APIs that connect to Studio and Studio Pro.

# External entities **(Studio** **Pro)**

All entities included in the Mendix module through the Data Hub Catalog pane are referred to as external entities in Studio/Studio Pro and represented in the Domain Model with purple containers instead of the Blue Mendix data entity representation. They will have the corresponding differences in the properties as given in the **external entity properties**. 

The major **characteristic** of an external entity is that it is a *connection* to the dataset in the “publishing app” which is accessed and used in the consuming app by end-users.  Currently the consumed data is used by the consuming app.

External entities are read-only, so it is not possible to change te structural values of attributes or associations between two external entities.

For full details see [external entities](ref??)

# Data Sources 

> Data sources are the apps/services which define the data that is being made available through it. The generic term used for the app, database or service which owns and provides access to a dataset that can be consumed in Studio/Studio Pro through external entities. From the point of view of the Studio/Studio Pro user, they are connecting through a data source when they use the external entity of this source and using the dataset of the selected entity.

Reviewer: ```Data sources are the apps/services``` can we say they're the services instead?

Author: Thanks for the feedback. The doc is very much is rough draft stage and not ready for review.

A service is a collection of datasets.

It is the endpoint.

# Datasets

To users, the **dataset**  data, regardless of the system “owning” the data.  A **dataset** is the data similar to a table or excel sheet, with the actual set of data that the user will connect to.

An **entity** is the modelling construct for a dataset - or more specifically in Mendix, the **entityset** - which is the connection to the available data.

> The **external entity** in the consuming app connects to the dataset that is maintained in the originating app via the published service (the API to the originating app) registered in the DHC. Currently datasets for external entities are updated in the source application of the data source. Data is retrieved on the initiative by the consuming app (pull-based), so Data Hub does not play a role in updating the consuming apps: the consuming apps receive the latest data when they “request it” through the API.

Reviewer: ```suggestion
The **external entity** in the consuming app connects to the dataset that is maintained in the originating app via the published service (the API to the originating app) registered in the Data Hub Catalog. Datasets for external entities are updated in the source application of the data source. Data is retrieved on the initiative by the consuming app (pull-based), so Data Hub does not play a role in updating the consuming apps: the consuming apps receive the latest data when they request it through the API.
```

{{% alert type="info" %}}
This is not compulsory, and users can continue to use an older version of a service unless the new version was deployed to the same service endpoint as the previous version. In Studio Pro, new versions of a service are indicated and users can choose to **Update** the service, or **Switch** to another version of the service deployed to another endpoint.
{{% /alert %}}

It is good practice that publishers of a service serve a notice of deprecation on a service version that will be replaced with a new service that may contain breaking changes which would cause the consuming app to fail. In this case the updated service should be deployed to a new service endpoint. In this case, in Studio Pro, users will get the option to **Switch** to the new version. 

# **Registration**

> This is the process of “adding” a shareable asset – data source (service) or dataset – to Data Hub. This is means that the metadata is available and can be searched/curated etc.

Reviewer: ```suggestion
This is the process of “adding” a shareable asset – data source (service) or dataset (entity) – to Data Hub. This is means that the metadata is available and can be searched/curated etc.
```

# Consume

Users connecting to the dataset that is made available or registered in Data Hub is  known as *Consume*. 

 A dataset is consumed in an app. Users can find the correct dataset by searching in the Catalog. They  **consume** in Studio Pro through the Data Hub Pane.

Users can also **consume** the dataset through the URL that is available for the asset in the asset details page or by using the DH API.

 Consume is of a specific dataset (entityset), of an endpoint, in a particular environment.

 

In Studio Pro, external data sources registered in the Data Hub Catalog and can be discovered through the **Data Hub Pane**  and introduced in the domain model as *external entities* which can be used with local entities with the difference that external entities **connect** to data that is maintained in the source application and therefore is **read or consumed**. Changes to this data is done in the originating apps.

> ## Entitysets

Reviewer: ```suggestion
## Entity sets
```

> To make data from your apps available to others, you can publish the datasets in an **OData service** and register it in the Data Hub. In a Mendix app, the datasets are the **Entity sets** for a defined **Entity**. Further details such as the location of the registered dataset are also captured: for Mendix app deploying to the Mendix cloud this information is taken during the deployment process. 

Reviewer: ```suggestion
To make data from your apps available to others, you can publish the datasets in an **OData service** and register it in the Data Hub. In a Mendix app, the datasets are the **Entity sets** for a defined **Entity**. Further details such as the location of the registered dataset are also captured: for a Mendix app deploying to the Mendix Cloud this information is captured during the deployment process. 
```

Deployment [environments](/developerportal/deploy/environments) should be clearly defined for the organization. The environment that an app is deployed to is important and indicates the quality and reliability of the app and data. 

## Deployment

## Published OData Service

At the service endpoint are the OData service metadata contract files that define the data exposed by the service. These metadata contracts define the exposed attributes and associations and the links for connecting to the data they define. 

The contract of the published OData service (the *$metadata* document) contains the details of what is exposed in the service. This includes the metadata of the exposed datasets (or entity sets in Mendix Studio Pro) and their exposed attributes, associations, types, and accessibility. This information about the registered OData service is shown in the search details. 

# Supported OData Protocols

OData v3 (Mendix only)

OData v4 (Other Business applications)

## Service Versions

> Every published OData service  has a version number, and apps that consume a datasource will consume from a specific version. Updates and changes to a service will be indicated by a change in the version number if good practice is followed by the data source originators. This means that there may be several versions of a registered data source available in the catalog and they will all be listed as separate items in the search results for the same-named data source.

Reviewer: ```suggestion
Every published OData service  has a version number, and apps that consume a data source will consume from a specific version. Updates and changes to a service will be indicated by a change in the version number if good practice is followed by the data source originators. This means that there may be several versions of a registered data source available in the catalog and they will all be listed as separate items in the search results for the same-named data source.
```

---
title: "Concepts behind Mendix Data Hub and Data Sharing (Working title)"
category: "General Info"
menu_order: 12
description: "Describes terminology and concepts"
tags: ["data hub catalog", "data hub", "virtual entities", "landscape", "published odata service"]

---

# 1 Introduction

Describes concepts behind using the Mendix Data Hub and sharing datasets.

Data Hub makes it simpler to use data between Apps. Catalog allows governance of which data is used by whom, and life cycle management informing people of changes, and with good versioning of OData contracts.

##  1.1 The Role of a Data Hub in an Organization

The Catalog is the hub in the collaboration, discovery and connection of an organizations data source.   It catalogs microservices from the organization's business applications systems using OData contracts. The Data Hub can already import OData contracts from 3rd party systems such as Teamcenter, SAP & Microsoft and OData v4services from other business applications.

> The Data Hub facilitates an integration mechanism, where architects and lead developers can define data that is available and what the data means. Citizen Developers?? can use the data easily, ??and where impact analysis for who uses which data is out of the box??.

Reviewer 2: ```suggestion
The Data Hub facilitates an integration mechanism, where architects and lead developers can define data that is available and what the data means. Citizen Developers?? can use the data easily, ??and where impact analysis for who uses which data is out of the box?.
```
